dnd-troops-service/src/main/java/cz/muni/fi/pa165/w2018/dndtroops/service/facade/UserFacadeImpl.java

```diff
@@ -0,0 +1,81 @@
+package cz.muni.fi.pa165.w2018.dndtroops.service.facade;
+
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.UserAuthenticateDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.dto.UserDTO;
+import cz.muni.fi.pa165.w2018.dndtroops.api.facade.UserFacade;
+import cz.muni.fi.pa165.w2018.dndtroops.backend.entity.User;
+import cz.muni.fi.pa165.w2018.dndtroops.service.BeanMappingService;
+import cz.muni.fi.pa165.w2018.dndtroops.service.UserService;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+public class UserFacadeImpl implements UserFacade{
+    @Inject
+    private UserService userService;
+
+    @Inject
+    private BeanMappingService mapper;
+
+    @Override
+    public UserDTO getById(long id) {
+        return mapper.mapTo(userService.getById(id), UserDTO.class);
+    }
+
+    @Override
+    public UserDTO getByLogin(String login) {
+        return mapper.mapTo(userService.getByLogin(login), UserDTO.class);
+    }
+
+    @Override
+    public List<UserDTO> getAll() {
+        return mapper.mapTo(userService.getAll(), UserDTO.class);
+    }
+
+    @Override
+    public long createUser(UserDTO user, String plainPassword) {
+        User usr = mapper.mapTo(user, User.class);
+        userService.createUser(usr, plainPassword);
+        usr = userService.getByLogin(user.getLogin());
```

Comment:
You don't have to do this, usr entity will have the id assigned by entity manager and can be use in return